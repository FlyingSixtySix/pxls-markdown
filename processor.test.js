const { setProcessor, expectProcessorOutput } = require('./testUtil.js');

const { processor, plugins } = require('./index.js');

setProcessor(
	processor()
		.use(plugins.emoji, {
			emojiDB: { dog: '🐶', grinning: '😁' },
			emojiRegex: /🐶|😁/i
		})
		.use(plugins.methodWhitelist, {
			block: [ 'blankLine' ],
			inline: [ 'coordinate', 'emoji_raw', 'emoji_name', 'mention', 'escape', 'autoLink', 'url', 'underline', 'strong', 'emphasis', 'deletion', 'code' ]
		})
);

test('renders everything correctly', () => {
	expectProcessorOutput(`
	*italic* _italic_
	**bold**
	__underline__
	~~strikethrough~~
	\`inline code\`
	https://a.link
	@mention
	:dog: 😁
	(1234, 567, 50x)
	\\*escaping* \\_any_ \\**markup** \\__like__ \\~~this~~ (\\\`with\` \\:a: \\@backslash)
	`).toMatchSnapshot();
})
