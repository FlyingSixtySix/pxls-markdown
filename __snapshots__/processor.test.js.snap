// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders everything correctly 1`] = `
Array [
  "	",
  <i>
    italic
  </i>,
  " ",
  <i>
    italic
  </i>,
  "
	",
  <b>
    bold
  </b>,
  "
	",
  <u>
    underline
  </u>,
  "
	",
  <s>
    strikethrough
  </s>,
  "
	",
  <code>
    inline code
  </code>,
  "
	",
  <a
    href="https://a.link"
    target="_blank"
  >
    https://a.link
  </a>,
  "
	",
  <span
    class="mention"
  >
    @mention
  </span>,
  "
	",
  <span
    title=":dog:"
  >
    🐶
  </span>,
  " ",
  <span
    title=":grinning:"
  >
    😁
  </span>,
  "
	",
  <a
    href="#x=1234&y=567&scale=50"
  >
    (1234, 567, 50x)
  </a>,
  "
	",
  "*",
  "escaping* ",
  "_",
  "any_ ",
  "*",
  "*markup** ",
  "_",
  "_like__ ",
  "~",
  "~this~~ (",
  "\`",
  "with\` ",
  ":",
  "a: ",
  "@",
  "backslash)
	",
]
`;
